docs/dev-notes/BACKGROUND-QUEUE.md

```diff
@@ -89,8 +87,50 @@ C. push metric opened for push notification Z
+1. Save the first name to local storage such as a database to act as a cache. This changes the local state of the app. 
+2. Add a task to the background queue to eventually send the new first name to the remote API. It is at this time that you *provide to the background queue the new first name value* to be sent to the remote API. It's like you are taking a snapshot of the state of the SDK at this time and giving it to the background queue. You do not want to query the app's storage for the new first name when the background queue runs. You want to provide the actual first name value to the queue when you add the task. 
+
+```swift
+let profileId = 5 // the identifier to query our app's local data storage for this profile
+let newFirstName = "Eddie"
+database.updateFirstName(id: profileId, firstName: newFirstName)
+
+// Do this:
+backgroundQueue.addTask(
+    "edit_profile",
+    EditProfileQueueTaskData(newFirstName)
+)
+
+// Do *not* do this:
+backgroundQueue.addTask(
+    "edit_profile",
+    EditProfileQueueTaskData(profileId) // when "edit_profile" task runs in the background queue, query the database for the first name for id 5
+)
+```
+
+Why this is important is to avoid a scenario like this:
+* Your first name in app is `Dana`. 
+* You change your first name to `Eddie`. Add adds task to background queue to update first name. 
+* Background queue task has not yet run yet. You change your first name to `Frankie`. Add task to background queue to update first name. 
+* Background queue now has 2 tasks in it. The queue runs task 1 to update first name. If you provided the value `Eddie` to the background queue at the time of adding the event to the queue, the queue at this point would know to send the value `Eddie` to the API. But if you didn't provide `Eddie` and you are instead relying on your app simply checking the current state of the app's cached data, the background queue would instead send the value `Frankie` to the API and would never send `Eddie`. Your remote API would never know about the value `Eddie`. 
+
+For some use cases like the one above, this should not impose a threat to the state of your app or it's data. But what if you have a different use case that involves deleting? Such as a Twitter app allowing you to send tweets. If the following events happen in your app:
+* Phone in airplane mode. User creates a new tweet. Task added to background queue to send tweet in the local database with ID 5 to remote API. 
+* Phone still in airplane mode and user decides to delete tweet. Task added to background queue to delete tweet. App deletes row in database with ID 5. 
+* Phone not in airplane and background queue runs. Queue task sees no row in database for ID 5, decides not to run. 
+* .....what do you do now? The state of the background queue is now out of sync with your app's events? 
```

Comment:
i think one scenario is enough to get the point across so i'd vote to get remove this scenario all together